lib/filterx/tests/test_scope.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024 Axoflow
+ * Copyright (c) 2024 László Várady
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <criterion/criterion.h>
+
+#include "syslog-ng.h"
+#include "scratch-buffers.h"
+#include "libtest/filterx-lib.h"
+#include "apphook.h"
+#include "logmsg/logmsg.h"
+
+#include "filterx/filterx-scope.h"
+#include "filterx/filterx-variable.h"
+#include "filterx/object-primitive.h"
+
+Test(filterx_scope, test_scope_on_heap)
+{
+  FilterXScope *s = filterx_scope_new(NULL);
+  cr_assert_null(filterx_scope_lookup_variable(s, filterx_map_varname_to_handle("var", FX_VAR_DECLARED_FLOATING)));
+  filterx_scope_free(s);
+}
+
+Test(filterx_scope, test_scope_stacking)
+{
+  gsize alloc_size = filterx_scope_get_alloc_size();
+
+  FilterXScope *s = g_alloca(alloc_size);
+  filterx_scope_init_instance(s, filterx_scope_get_alloc_size(), NULL);
+
+  FilterXVariableHandle var = filterx_map_varname_to_handle("var", FX_VAR_DECLARED_FLOATING);
+  filterx_scope_register_variable(s, FX_VAR_DECLARED_FLOATING, var);
+
+  FilterXScope *s2 = g_alloca(alloc_size);
+  filterx_scope_init_instance(s2, filterx_scope_get_alloc_size(), s);
+
+  cr_assert_not_null(filterx_scope_lookup_variable(s2, var));
+
+  filterx_scope_clear(s2);
+  filterx_scope_clear(s);
+}
+
+Test(filterx_scope, test_scope_sync)
+{
+  gsize alloc_size = filterx_scope_get_alloc_size();
+  FilterXScope *s = g_alloca(alloc_size);
+  filterx_scope_init_instance(s, filterx_scope_get_alloc_size(), NULL);
+  LogMessage *msg = log_msg_new_empty();
+
+  filterx_scope_set_message(s, msg);
+
+  FilterXVariableHandle var = filterx_map_varname_to_handle("$var", FX_VAR_MESSAGE_TIED);
```

Comment:
Iirc I removed the need to start these names with a dollar. Actually this works as if you used $$var in the config

Reply:
`filterx_map_varname_to_handle()` removed the `$`.

Reply:
Maybe that patch is not yet merged though